### PR TITLE
Publish initial performance numbers

### PR DESCRIPTION
**Pull request by flands**
Addresses #276

### CODE REVIEW

File: PERFORMANCE.MD [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+has a few configurable parameters which can impact the results of the tests. The
+parameters used are defined below.
+
+* FlushInterval(ms) [default: 1000]
+* MaxQueueSize [default: 100]
+* SubmissionRate(spans/sec): 100,000
+
+## Results without tail-based sampling
+
+| Span<br>Format    | CPU<br>(2+ GHz) | RAM<br>(GB) | Sustained<br>Rate | Recommended<br>Maximum |
```
Comment (bogdandrutu) [inline PERFORMANCE.MD]: 1. Can we please add numbers with our own format? :)
1. What is the exporter in all these case?

Comment (pjanotti) [reply PERFORMANCE.MD]: 1. We still need to change the synthetic-load-generator to emit oc (although we could measure oc-agent -> oc-collector)
2. In this case we are interested on ingress rate so we measure with a "null exporter" (i.e.: --debug-processor)

Comment (bogdandrutu) [reply PERFORMANCE.MD]: We can do the following:

(1)load generator (import_format) -> (2)"test_instance" (export_fromat) -> (3)oc_agent with debug

where:
(1) is the load generator
(2) is the instance we monitor
(3) acts just as a backend (in reality they may have issues with scaling the backend, but this is beyond this project scope).

File: PERFORMANCE.MD [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+OpenCensus Service, are applicable primarily to the OpenCensus Collector and
+are measured only for traces. In the future, more configurations will be tested.
+
+Note with the OpenCensus Agent you can expect as good if not better performance
+with lower resource utilization. This is because the OpenCensus Agent does not
+today support features such as batching or retries and will not support
+tail-sampling.
+
+It is important to note that the performance of the OpenCensus Collector depends
+on a variety of factors including:
```
Comment (songy23) [inline PERFORMANCE.MD]: Also mention OS platform here?

File: PERFORMANCE.MD [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+* The receiving format: Jaeger thrift (14268) or Zipkin v2 JSON (9411)
+* The size of the spans (tests are based on number of tags): 20
+* Whether tail-sampling is enabled or not
+* CPU / Memory allocation
+
+## Testing
```
Comment (songy23) [inline PERFORMANCE.MD]: Maybe also specify which OS you're using when running the tests?

File: PERFORMANCE.MD (hunk 1 above)
Comment (flands) [reply PERFORMANCE.MD]: In order to get these numbers published today we will skip, but the goal is to include these going forward (ideally by using the Java ocagent exporter)

**Comment by flands** [comment]
Filed https://github.com/census-instrumentation/opencensus-service/issues/438

File: Makefile [hunk 4]
```diff
@@ -48,7 +48,7 @@ lint:
 
 .PHONY: install-tools
 install-tools:
-	go get -u golang.org/x/lint/golint
```
Comment (bogdandrutu) [inline Makefile]: The -u flag instructs get to use the network to update the named packages and their dependencies. By default, get uses the network to check out missing packages but does not use it to look for updates to existing packages.

I don't think removing this is necessary good. Maybe we should file an issue to restore this after lint fixes?

Comment (flands) [reply Makefile]: SGTM

Comment (pjanotti) [reply Makefile]: Yes, the idea is that this is a temporary workaround.

Comment (pjanotti) [reply Makefile]: #439

File: PERFORMANCE.MD (hunk 1 above)
Comment (yurishkuro) [reply PERFORMANCE.MD]: Please define what sustained rate means

Comment (flands) [reply PERFORMANCE.MD]: @yurishkuro done: https://github.com/census-instrumentation/opencensus-service/pull/441

Comment (yurishkuro) [reply PERFORMANCE.MD]: sorry, still not clear, that change just says "running for a minimum of one hour (i.e. sustained rate)". Rate of what? Number of spans received by the collector? What does it do with the spans besides receiving them?

**Comment by bogdandrutu** [comment]
Hi all,

Can we get a [pprof](https://golang.org/doc/diagnostics.html#profiling) sample while we run the loadtests. I am very interested to see where we have the performance problems - if possible get cpu/block/mutex (all would be great).

**Comment by flands** [comment]
We did not have pprof initially, but when we run the next set of tests we will.

**Pull request merged**